### PR TITLE
Lottery functionality

### PR DESCRIPTION
**Pull request by ludapops**
This Pr adds functionality to lottery page that works with the smart contract

### CODE REVIEW

File: apps/web/src/utils/contractHelpers.ts [hunk 1]
```diff
@@ -200,7 +200,7 @@ export const getPointCenterIfoContract = (signer?: Signer | Provider) => {
 export const getCakeContract = (signer?: Signer | Provider, chainId?: number) => {
   return getContract({
     abi: cakeAbi,
-    address: chainId ? CAKE[chainId].address : CAKE[ChainId.BSC].address,
+    address: chainId ? vertoTokens.verto.address : vertoTokensTestnet.verto.address,
```
Comment (yuushazen) [inline apps/web/src/utils/contractHelpers.ts:203]: This should be be checking if its mainnet, not checking for a chainId, please look at other places in the code where we do this,I think its `IS_MAINNET` or something like that

File: apps/web/src/views/Lottery/components/PreviousRoundCard/FooterExpanded.tsx [hunk 2]
```diff
@@ -44,25 +48,25 @@ const PreviousRoundCardFooter: React.FC<
     prizeInBusd = amountCollectedInCake.times(cakePriceBusd)
   }
 
-  const getTotalUsers = (): string => {
-    if (!lotteryGraphDataFromState && fetchedLotteryGraphData) {
-      return fetchedLotteryGraphData?.totalUsers?.toLocaleString()
-    }
+  // const getTotalUsers = (): string => {
```
Comment (yuushazen) [inline apps/web/src/views/Lottery/components/PreviousRoundCard/FooterExpanded.tsx:51]: Remove?

File: apps/web/src/views/Lottery/components/Slider.tsx [hunk 3]
```diff
@@ -91,6 +143,87 @@ const SlideContent = ({ title, date }) => {
...
+
+    return () => clearInterval(timer.current)
+  }, [selectedRoundId, currentLotteryId, numRoundsFetched, dispatch])
+
+  const handleInputChange = event => {
+    const {
+      target: { value },
+    } = event
+    if (value) {
+      setSelectedRoundId(value)
```
Comment (yuushazen) [inline apps/web/src/views/Lottery/components/Slider.tsx:207]: Assign a variable and then call `setSelectedRoundId`, don't call `setSelectedRoundId` multiple times

File: apps/web/src/views/Lottery/components/ViewTicketsModal/PreviousRoundTicketsInner.tsx [hunk 4]
```diff
@@ -174,12 +174,12 @@ const PreviousRoundTicketsInner: React.FC<React.PropsWithChildren<{ roundId: str
...
       </TopBox>
       <ScrollBox>
-        <Text bold textTransform="uppercase" color="secondary" fontSize="12px" my="16px">
+        <Text bold textTransform="uppercase" color="primary" fontSize="12px" my="16px">
           {t('Your tickets')}
         </Text>
         <Flex mb="8px" justifyContent="space-between">
           <Flex>
-            <Ticket width="24px" height="24px" mr="8px" />
+            {/* <Ticket width="24px" height="24px" mr="8px" /> */}
```
Comment (yuushazen) [inline apps/web/src/views/Lottery/components/ViewTicketsModal/PreviousRoundTicketsInner.tsx:182]: Remove?

**Review by yuushazen — CHANGES_REQUESTED** [review]

File: apps/web/src/views/Lottery/components/Slider.tsx (hunk 3 above)
Comment (ludapops) [reply apps/web/src/views/Lottery/components/Slider.tsx:207]: removed this parent function since it was unused

**Review by yuushazen — APPROVED** [review]

**Pull request merged**